core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -184,7 +201,15 @@ public boolean accept(CarbonFile file) {
         folderDetails.setStatus(SegmentStatus.SUCCESS.getMessage());
         setIndexFileNamesToFolderDetails(folderDetails, carbonFiles);
         segmentFile.addPath(location, folderDetails);
-        String path = writePath + "/" + CarbonUtil.generateUUID() + CarbonTablePath.SEGMENT_EXT;
+        String path = null;
+        if (isMergeIndexFlow) {
```

Comment:
i think its better to add comments in all the places where the merge index will be false saying that the code will be used for developer debugging purpose, so that it wont be confused with actual flow.

Reply:
added comments